### PR TITLE
Fixes #2 (depends on libstorj #382)

### LINKED ISSUE #2: Ruby's has a small RLIMIT_MEMLOCK and `LibStorj::Env` doesn't unlock atm
After about 9 consecutive new `LibStorj::Env` instances we start getting instances whose `#storj_env.to_ptr` points to NULL:

screenshot from 2017-08-18 12-42-18

I committed the tests with group labels (and logging) on the [wtf-help-plz](https://github.com/Storj/ruby-libstorj/blob/wtf-help-plz/spec/env_spec.rb) branch to easily repeat:
```bash
git fetch <remote>
git checkout <remote>/4bef4a038fb8330cb50a383f7859636abd374fa9 #wtf-help-plz

rspec -f doc -t working       #should pass 8 tests
rspec -f doc -t broken        #should pass 8 and fail 1 test
rspec -f doc -t extra_broken  #should pass 8 and fail 3 tests
```

### PR DESCRIPTION
**Pull request by bryanchriswhite**
#### Depends on [libstorj #382](https://github.com/Storj/libstorj/pull/382)

Adds `LibStorj::Ext::Storj::Env#destroy` which binds to the newly exposed `storj_destroy_env`.

### CODE REVIEW

File: lib/ruby_libstorj/env.rb [hunk 1]
```diff
@@ -13,8 +13,12 @@ class Env
     end
 
     def initialize(*options)
-      @pointer = ::LibStorj::Ext::Storj.method(:init_env).call(*options)
-      @pointer[:loop] = ::Libuv::Ext.default_loop
+      @storj_env = ::LibStorj::Ext::Storj.method(:init_env).call(*options)
+      @storj_env[:loop] = ::Libuv::Ext.default_loop
```
Comment (braydonf) [inline lib/ruby_libstorj/env.rb]: Shouldn't need to set this now, since it'll be the default loop

Comment (bryanchriswhite) [inline lib/ruby_libstorj/env.rb]: I get a segfault if i comment out that line... i think libuv is loaded twice and they're different default loops...

Comment (braydonf) [inline lib/ruby_libstorj/env.rb]: hmm... okay

Comment (bryanchriswhite) [inline lib/ruby_libstorj/env.rb]: u think that is going to cause issues? everything in libstorj references off of `env->loop` right?

Comment (braydonf) [inline lib/ruby_libstorj/env.rb]: It may not, having the loop configurable helps there.

Comment (braydonf) [inline lib/ruby_libstorj/env.rb]: Yeah, everything in libstorj references `env->loop`

Comment (bryanchriswhite) [inline lib/ruby_libstorj/env.rb]: alternatively there is a way to tell ruby-libuv to use a different loop which i think can be passed as a pointer wrapper.

**Comment by braydonf** [comment]
Tests are working for me with:
 ```
Finished in 0.33002 seconds (files took 0.16762 seconds to load)
13 examples, 0 failures
```

File: lib/ruby_libstorj/env.rb (hunk 1 above)
Comment (bryanchriswhite) [inline lib/ruby_libstorj/env.rb]: alright - so just out of curiosity i checked and i think ruby-libuv doesn't support changing the internal loop by out of the box... we could tweak it in our binding library but unless there's something specific that you think we should avoid by doing that, I'd say that setting this way is ideal.

could you plz merge if you're good with it :D

**Pull request merged**